### PR TITLE
Emelie Lindblom - project-express-api

### PR DESCRIPTION
**Pull request by waliem**
link to the deployed version: 
https://emelies-express-api.herokuapp.com/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,24 @@
 # Express API Project
 
-Replace this readme with your own information about your project. 
+This is a backend project with a purpose to build an API with hardcoded data in a json-file. And create endpoints and using a RESTful approach. The Data used is data over netflix titles both movies and tv shows.
```
Comment (isomoth) [inline README.md:3]: This is a very descriptive, to-the-point presentation of what your app does. Good one!

File: README.md [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
 
-## The problem
+- /endpoints = a overview of the endpoints by imported express-list-endpoints
+- /netflix-titles = returning the whole array or data avalible
+- /netflix-titles/movies endpoint that returns only movie titles using filter
+- /netflix-titles/tvshows endpoint that returns only tvshow titles using filter
+- /netflix-titles/year/:year that makes it possible to search and filter the response total data array by a specific year
+- /netflix-titles/show_id/:show_id that makes it possible to find a specific movie or tv show based on a unique show id
+- /netflix-titles/title/:title that makes it possible to search for a title by a specific title through params
+- /netflix-titles?title={title of the movie} endpoint that makes it possible to search title through query
```
Comment (isomoth) [inline README.md:14]: Documenting the endpoints in the README is really useful. I got inspired by this!

File: server.js [hunk 3]
```diff
@@ -22,11 +16,106 @@ app.use(cors())
 app.use(express.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+app.get("/", (req, res) => {
+  res.send(
+    "Hello This is the root-endpoint to my netflix API. By adding /endpoints to the url you can get an overview of the possible endpoints"
```
Comment (isomoth) [inline server.js:22]: This is a very user-friendly default route. You're really user-oriented!

File: server.js [hunk 4]
```diff
@@ -22,11 +16,106 @@ app.use(cors())
...
+  let titles = netflixData
+
+  if (title) {
+    titles = titles.filter(
+      (item) => item.title.toLowerCase().indexOf(title.toLowerCase()) !== -1
+    )
+  } else {
+    res.status(200).json({
+      response: netflixData,
+      success: true,
```
Comment (isomoth) [inline server.js:42]: You use error handling quite consistently throughout your API. Good job!

**Review by isomoth — APPROVED** [review]
Good job Emelie! Your deployed version works just like it should and is quite easy to navigate thanks to your detailed documentation. 

I especially liked your user-oriented mindset when working with the backend. Verbal references become really important for the user in these cases, and you managed them quite well! I like how you not only give really clear instructions, but also write small details like "Hello" and "Sorry" that make your app more personal. 

Keep it up!

File: server.js (hunk 4 above)
Comment (waliem) [reply server.js:42]: Thank you for the review, Isabel!

**Pull request closed without merging**